templates/commands/render_test.go

```diff
@@ -34,6 +35,27 @@ import (
 	"github.com/hashicorp/go-getter/v2"
 )
 
+// cmpFileMode is a cmp option that handles the conflict between Unix and
+// Windows systems file permissions.
+var cmpFileMode = cmp.Comparer(func(a, b fs.FileMode) bool {
+	// Windows really only has 2 file modes: 0666 and 0444[1]. Thus we only check
+	// the first bit, since we know the remaining bits will be the same.
+	// Furthermore, there's no reliable way to know whether the executive bit is
+	// set[2], so we ignore it.
+	//
+	// I tried doing fancy bitmasking stuff here, but apparently umasks on Windows
+	// are undefined(?), so we've resorted to substring matching - hooray.
+	//
+	// [1]: https://medium.com/@MichalPristas/go-and-file-perms-on-windows-3c944d55dd44
+	// [2]: https://github.com/golang/go/issues/41809
+	if runtime.GOOS == "windows" {
+		// "-rw-r--r--"
```

Comment:
Super-nit: the very helpful comment above explains that the possible modes are 0666 and 0444, but this example is neither (0644).